core/account/builder.go

```diff
@@ -109,27 +108,34 @@ func (m *Manager) DecodeSpendUTXOAction(data []byte) (txbuilder.Action, error) {
 
 type spendUTXOAction struct {
 	accounts *Manager
-	TxHash   *bc.Hash `json:"transaction_id"`
-	TxOut    *uint32  `json:"position"`
+	OutputID *bc.OutputID `json:"output_id"`
+	TxHash   *bc.Hash     `json:"transaction_id"`
+	TxOut    *uint32      `json:"position"`
 
 	ReferenceData chainjson.Map `json:"reference_data"`
 	ClientToken   *string       `json:"client_token"`
 }
 
 func (a *spendUTXOAction) Build(ctx context.Context, b *txbuilder.TemplateBuilder) error {
-	var missing []string
-	if a.TxHash == nil {
-		missing = append(missing, "transaction_id")
-	}
-	if a.TxOut == nil {
-		missing = append(missing, "position")
-	}
-	if len(missing) > 0 {
-		return txbuilder.MissingFieldsError(missing...)
-	}
+	var outid bc.OutputID
 
-	out := bc.Outpoint{Hash: *a.TxHash, Index: *a.TxOut}
-	res, err := a.accounts.utxoDB.ReserveUTXO(ctx, out, a.ClientToken, b.MaxTime())
+	// This is compatibility layer - legacy apps can spend outputs via the raw <txid:index> pair.
+	// If none of the parameters are provided we need to report that output_id is missing.
+	if a.TxHash != nil || a.TxOut != nil {
+		if a.TxHash == nil {
+			return txbuilder.MissingFieldsError("transaction_id")
+		}
+		if a.TxOut == nil {
+			return txbuilder.MissingFieldsError("position")
+		}
+		outid = bc.ComputeOutputID(*a.TxHash, *a.TxOut)
+	} else {
+		if a.OutputID == nil {
+			return txbuilder.MissingFieldsError("output_id")
+		}
+		outid = *a.OutputID
+	}
```

Comment:
That's harder to parse what's being meant here... But I get the point, how about this flat list of 3 options: https://github.com/chain/chain/pull/417/commits/36ad7c07b41505885c772102e72406a36ea4d402